### ISSUE TITLE
Client/Server className mismatch in Next.JS

### ISSUE THREAD

**Issue by ellsclytn** [jacobwgillespie/styled-icons#452]
I've got a minimal reproduction of the issue here: https://github.com/ellsclytn/next-styled-icons

When using Styled Icons with Next.JS (and the associated with-styled-icons plugin), hot reloading works, but causes a browser console error and warning:

Error:

```
index.js:2178 Warning: Prop `className` did not match. Server: "sc-hbkdQL pages__MyIcon-sc-1wbo52z-1 KKtVi" Client: "sc-jzgbtB pages__MyIcon-sc-1wbo52z-1 KKtVi"
```

Warning:

```
styled-components.browser.esm.js:1927 It looks like you've wrapped styled() around your React component (svg), but the className prop is not being passed down to a child. No styles will be rendered unless className is composed within your React component.
```

This does not occur on an initial load immediately after server start, but rather only occurs when you perform a manual browser reload after making a change to the code. You can replicate this by commenting/uncommenting line 17 of pages/index.js and refreshing the page each time.

**Comment by jacobwgillespie** [comment]
Hey, so I spent some time digging into this and was not able to determine the cause.  My guess is that after hot reloading, the content passed to Styled Components is different than before, so it creates a new ID for the component.  Basically the build pipeline differs slightly somehow between the client and server, hence the className mismatch.

There are quite a few issues in the Next and Styled Components repos about this, but none seemed to match up quite right to this particular case.

Good news is that this appears to only affect Next development mode, and production builds are unaffected.  And as you said, hot reloading does still function.

But it's an annoyance - I'll try to do some more investigation when I have time, and/or if anybody else wants to take a look that would be appreciated!

**Comment by ellsclytn** [comment]
Thanks for looking into it. I agree, not a *massive* problem given that it's merely in development mode that it gets seen.

Seeing as I'm using this in a fairly big project, I'll probably see it from time to time. In which case, I'll keep an eye out for potential solutions too

**Comment by jacobwgillespie** [comment]
@ellsclytn I believe I found the fix, which has been released as v6.0.0! 🎉 

The issue seems to be that if you have `babel-plugin-styled-components` in your Babel config, it only applies to components created in your project, not in libraries (code in `node_modules`).  That's fine for dependencies generally, but if you wrap a Styled Icon component in `styled()` the SC component ID for the base icon isn't stable across server-side rendering updates.

The fix appears to be that I need to process all the Styled Icons through `babel-plugin-styled-icons` before publishing, which assigns them all a stable component ID.

Let me know if v6.0.0 works for you! 🎄
